trade-finance-manager-api/api-tests/v1/deal-cancellation/submit-deal-cancellation.post.api-test.ts

```diff
@@ -0,0 +1,154 @@
+  bankRequestDate: new Date().valueOf(),
+  effectiveFrom: new Date().valueOf(),
+});
+
+describe('POST /v1/deals/:id/cancellation/submit', () => {
+  let testUsers: Awaited<ReturnType<typeof initialiseTestUsers>>;
+  let aPimUser: TestUser;
+
+  beforeAll(async () => {
+    testUsers = await initialiseTestUsers(app);
+    aPimUser = testUsers().withTeam(TEAM_IDS.PIM).one();
+  });
+
+  describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is disabled', () => {
+    beforeEach(() => {
+      process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'false';
+    });
+
+    afterAll(() => {
+      process.env = originalProcessEnv;
+    });
+
+    it('returns a 404 response for an authenticated user with a valid id path', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: validId });
+
+      // Act
+      const response = await as(aPimUser).post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(404);
+    });
+  });
+
+  describe('when FF_TFM_DEAL_CANCELLATION_ENABLED is enabled', () => {
+    beforeEach(() => {
+      process.env.FF_TFM_DEAL_CANCELLATION_ENABLED = 'true';
+    });
+
+    afterAll(() => {
+      process.env = originalProcessEnv;
+    });
+
+    withTeamAuthorisationTests({
+      allowedTeams: [TEAM_IDS.PIM],
+      getUserWithTeam: (team) => testUsers().withTeam(team).one(),
+      makeRequestAsUser: (user: TestUser) =>
+        as(user)
+          .post(aValidPayload())
+          .to(getSubmitTfmDealCancellationUrl({ id: validId })),
+      successStatusCode: 200,
+    });
+
+    it('returns a 401 response when user is not authenticated', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: validId });
+
+      // Act
+      const response = await post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(401);
+    });
+
+    it('returns a 400 response when the id path param is invalid', async () => {
+      // Arrange
+      const url = getSubmitTfmDealCancellationUrl({ id: 'invalid' });
+
+      // Act
+      const response = await as(aPimUser).post(aValidPayload()).to(url);
+
+      // Assert
+      expect(response.status).toEqual(400);
+    });
+
+    const invalidPayloads = [
+      {
+        description: 'the reason is not a string',
+        payload: { ...aValidPayload(), reason: 12 },
+      },
+      {
+        description: `the reason is over ${MAX_CHARACTER_COUNT} characters`,
+        payload: { ...aValidPayload(), reason: 'x'.repeat(MAX_CHARACTER_COUNT + 1) },
+      },
+      {
+        description: 'the reason is undefined',
+        payload: { ...aValidPayload(), reason: undefined },
+      },
+      {
+        description: 'the bankRequestDate is a string ',
+        payload: { ...aValidPayload(), bankRequestDate: new Date().toISOString() },
+      },
+      {
+        description: 'the bankRequestDate is undefined ',
+        payload: { ...aValidPayload(), bankRequestDate: undefined },
```

Comment:
I think in such POST validation tests, instead of undefined, it should be an empty string `''`, since this is what `req.body` would contain if nothing is provided - same on line 112 and 128,

Reply:
Same in the middleware

Reply:
I dont think I quite understand what you mean
> since this is what req.body would contain if nothing is provided
Its possible for the payload to not include a property, so it would be undefined

Reply:
A typical use case would be submitting an empty form, or the form with valid values apart from `bankRequestDate`. In this case it'd come through as `bankRequestDate: ''`

The only way AFAIK It would be `undefined` is if a user is manipulating the DOM or network request

I'd prefer this to be an empty string rather than specifically `undefined`. What do you think?

Reply:
Sorry @MarRobSoftwire I was thinking this was UI POST test 🤦 

Happy with this but should we add a test for if `bankRequestDate` is an empty string?

Reply:
I've got a test
```
    {
      description: "the 'bankRequestDate' is a string",
      payload: {
        ...aValidPayload(),
        bankRequestDate: new Date().toString(),
      },
    },
```
The zod validation is looking for bankRequestDate to be a number so i dont think we gain anything new by checking for an empty string?

Reply:
OK makes sense, thanks!